core/src/test/scala/org/apache/spark/storage/MemoryStoreSuite.scala

```diff
@@ -51,12 +52,34 @@ class MemoryStoreSuite
   implicit def StringToBlockId(value: String): BlockId = new TestBlockId(value)
   def rdd(rddId: Int, splitId: Int): RDDBlockId = RDDBlockId(rddId, splitId)
 
+  // Save modified system properties so that we can restore them after tests.
+  val originalArch = System.getProperty("os.arch")
+  val originalCompressedOops = System.getProperty(TEST_USE_COMPRESSED_OOPS_KEY)
+
+  def reinitializeSizeEstimator(arch: String, useCompressedOops: String): Unit = {
+    val set = (k: String, v: String) => {
+      if (v == null) {
+        System.clearProperty(k)
+      } else {
+        System.setProperty(k, v)
+      }
+    }
+    set("os.arch", arch)
+    set(TEST_USE_COMPRESSED_OOPS_KEY, useCompressedOops)
+    val initialize = PrivateMethod[Unit](Symbol("initialize"))
+    SizeEstimator invokePrivate initialize()
+  }
+
   override def beforeEach(): Unit = {
     super.beforeEach()
     // Set the arch to 64-bit and compressedOops to true to get a deterministic test-case
-    System.setProperty("os.arch", "amd64")
-    val initialize = PrivateMethod[Unit](Symbol("initialize"))
-    SizeEstimator invokePrivate initialize()
+    reinitializeSizeEstimator("amd64", "true")
```

Comment:
Should this second arg be true here? Some tests that don't set this currently now set it to true, some to false.

Reply:
Yeah, this test suite assumes that compressed oops are enabled. The comment above alludes to this but the original code didn't actually override it. This leads to test failures when running a JVM where compressed oops are disabled.

In another test ("64-bit arch with no compressed oops") the test was inheriting the compressed oops setting of false from a preceding test rather than setting it itself. Since the preceding test is now cleaning up after itself that test now explicitly needs to set compressed oops to false.

In general it seems better to explicitly set these settings. Or we could try and make the tests more robust to different arch and compressed oops combinations. That is a bit more involved than this patch though.

Reply:
OK if they're all on purpose and as intended, that's fine.